aws_generated_data/commands/rds_eol.py

```diff
@@ -0,0 +1,158 @@
+import logging
+from collections.abc import Iterable
+from datetime import (
+    date,
+    datetime,
+    timedelta,
+)
+from pathlib import Path
+from typing import Any
+
+import requests
+import typer
+import yaml
+from bs4 import BeautifulSoup
+from pydantic import (
+    BaseModel,
+    RootModel,
+    ValidationError,
+)
+from typing_extensions import Annotated
+
+app = typer.Typer()
+log = logging.getLogger(__name__)
+
+
+class RdsItem(BaseModel):
+    engine: str
+    version: str
+    eol: date
+
+    def __lt__(self, other: Any) -> bool:
+        if not isinstance(other, RdsItem):
+            return False
+        return (self.engine, self.version) < (other.engine, other.version)
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, RdsItem):
+            return False
+        return (self.engine, self.version) == (other.engine, other.version)
```

Comment:
partial override `__eq__` require carefully checking, as `rds_item_a == rds_item_b` is `True` but they may have different `eol`, how about convert `rds_items` to dict for processing? Like this:

```python
    rds_items_dict = {
        (item.engine, item.version): item
        for item in rds_items
    }

    for engine in engines:
        log.info(f"Processing {engine} ...")
        for item in get_rds_eol_data(engine):
            rds_items_dict[(item.engine, item.version)] = item

    rds_items = filter_rds_items(
        rds_items_dict.values(),
        expired_date=date.today() - timedelta(days=clean_up_days),
    )
```

Reply:
Yeah, this is easier to read! 🙇 
Applied in [2dfceff](https://github.com/app-sre/aws-generated-data/pull/1/commits/2dfceff27779c1b51d379f4315e19aebf5dada0c)